Remotion/Web/Development.WebTesting/DownloadInfrastructure/DownloadedFileFinder.cs

```diff
@@ -108,71 +107,45 @@ public FileInformationTuple (DateTime lastWriteTimeUtc, long length)
       {
         var newFiles = GetNewFiles (filesInDownloadDirectoryBeforeDownload, startedDownloadHandlingUtc);
 
-         //The first step, we haven't found a partial File yet to monitor the download process
-        if (partialFileState == null)
+        if (TryGetPartialFile (newFiles, out var partialFile)
+            && !PartialFileWasFoundInPreviousIteration()
+            && TryGetFileInformation (partialFile, out var fileInfoOfFoundPartialFile))
         {
-          var partialFile = GetPartialFile (newFiles);
-          //Browser has started the download, remember the partial File for further monitoring
-          if (partialFile != null)
-          {
-            var fileInfo = GetFileInformation (partialFile);
-
-            if (fileInfo != null)
-            {
-              partialFileState = new PartialFileState (partialFile, fileInfo.LastWriteTimeUtc, fileInfo.Length);
-              currentStateTimeout = downloadUpdatedTimeout;
-            }
-          }
-          //No partial file found (yet) 
-          else if (
-              //Some browser create a temporary file shortly before creating a partial file, 
-              //so if that happens we know that a partial file is going to come
-              !_downloadFileFinderStrategy.ContainsPreDownloadFiles (newFiles)
-              //Try to find the downloaded file
-              && newFiles.Any())
-          {
-            var fileName = _downloadFileFinderStrategy.FindDownloadedFile (newFiles);
-
-            return new DownloadedFile (Path.Combine (_downloadDirectory, fileName), fileName);
-          }
+          _partialFileStateOfCurrentDownload = new PartialFileState (
+              partialFile,
+              fileInfoOfFoundPartialFile.LastWriteTimeUtc,
+              fileInfoOfFoundPartialFile.Length);
+          currentStateTimeout = downloadUpdatedTimeout;
         }
-        //We already have found a partial file which we can use to monitor the state of the download
-        else
+        else if (PartialFileExists (newFiles)
+                 && PartialFileWasFoundInPreviousIteration()
+                 && PartialFileWasUpdated()
+                 && TryGetFileInformation (_partialFileStateOfCurrentDownload.GetPartialFile(), out var fileInfoOfPartialFile))
+        {
+          _partialFileStateOfCurrentDownload.UpdatePartialFileLastWriteAccessUtc (fileInfoOfPartialFile.LastWriteTimeUtc);
+          _partialFileStateOfCurrentDownload.UpdatePartialFileLength (fileInfoOfPartialFile.Length);
+
+          downloadTimeWithoutUpdate.Restart();
+        }
+        else if (!PartialFileExists (newFiles) && !TemporaryFilesExist (newFiles) && newFiles.Count > 0 && newFiles.Any (IsZeroLengthFile))

```

Comment:
Isn't this equivalent to writing newFiles.Any(IsZeroLengthFile) ?

Reasoning: if any of the other condiitons is false, it will fall through the next if-else, too. 

It may still be a relevant bit of documentation, but that may be true, or not.

Reply:
I tried thinking about this but the reality is that without the other conditions, the tests fail

Reply:
That's...strange. something for the mob session, then.

Reply:
Yeah, this entire PR for me was "Let me make this code cleaner, oh wait, it doesn't work anymore" so I'd suggest I present you with a solution that is *stable* and *readable* and we take mental notes of the *optimal* way for the mob session 😄

Reply:
okay, the new code is different for the two branch conditions, so the original assumption no longer holds.